client/cli/ok.py

```diff
@@ -187,23 +187,39 @@ def main():
         # Instantiating assignment
         assign = assignment.load_assignment(args.config, args)
 
-        if args.authenticate:
-            # Authenticate and check for success
-            if not assign.authenticate(force=True):
-                exit(1)
-
         if args.tests:
             print('Available tests:')
             for name in assign.test_map:
                 print('    ' + name)
             exit(0)
 
-        msgs = messages.Messages()
-        for name, proto in assign.protocol_map.items():
-            log.info('Execute {}.run()'.format(name))
-            proto.run(msgs)
+        retry = True
+        while retry:
+            retry = False
+            if args.authenticate:
+                # Authenticate and check for success
+                if not assign.authenticate(force=True):
+                    exit(1)
 
-        msgs['timestamp'] = str(datetime.now())
+            try:
+                msgs = messages.Messages()
+                for name, proto in assign.protocol_map.items():
+                    log.info('Execute {}.run()'.format(name))
+                    proto.run(msgs)
+                msgs['timestamp'] = str(datetime.now())
+            except ex.AuthenticationException as e:
+                if not args.authenticate:
+                    args.authenticate = True
```

Comment:
nit: I'd prefer not to munge `args.authentication`. Is it possible to just create a local variable and reference that.

Reply:
I would like that too but the problem I saw is, what if the next committer forgets we have a different variable and tries to read that field of `args` directly? I wanted to guard against that somehow... any thoughts?

Reply:
+1 for not mutating args.argument.

Reply:
So I'm not sure what a better solution would be (even though I would similarly prefer `args` wasn't modified) so if you guys don't either I'm just going to merge soon...

Reply:
Can you just use a local variable instead of `args`?

Reply:
@colinschoen I can but it has the problem I mentioned above? https://github.com/okpy/ok-client/pull/356#discussion_r245890301

Reply:
@colinschoen actually on a second thought I don't think I can even do that, since `args.no_browser` is used in `get_code`, and it needs to see the change.